src/multupd.jl

```diff
@@ -148,7 +148,7 @@ function update_wh!(upd::MultUpdDiv{T}, s::MultUpdDiv_State{T}, X, W::Matrix{T},
     mul!(WtQ, transpose(W), Q)
     sum!(fill!(sW, 0), W)
     @inbounds for j = 1:n, i = 1:k
-        H[i,j] *= (WtQ[i,j] / sW[i])
+        H[i,j] *= (WtQ[i,j] / (sW[i] + lambda))
```

Comment:
I'm not particularly familiar with these algorithms. Is the regularization coefficient guaranteed to be small enough that it won't meaningfully change the values computed here?

Reply:
No, it is not guaranteed. I'm sorry, I didn't realize that `lambda` is the regularization coefficient. However, it seems that `lambda` is neither the L1 regularization coefficient nor the L2 regularization coefficient according to [1], [2]. So, I'll modify the code as follows:

- Delete `lambda` which has unclear meaning, and define new coefficients `lambda_w` and `lambda_h`
- Assume that `lambda_w` and `lambda_h` are the L1 regularization coefficient for W and H, respectively
- Define a new small positive constant `delta` to avoid division by zero 
- Assume that `delta` is not user-specified

Also, I'll create a test on non-negative constraints, NaN, and the accuracy of the solutions. 

[1] A. Cichocki, S. Amari, R. Zdunek, R. Kompass, G. Hori and Z. He, "Extended SMART algorithms for non-negative matrix factorization,"  Proceedings of the 8th International Conference on Artificial Intelligence and Soft Computing, pp.548–562, 2006.
[2] A. Koso and N. Takahashi, "Derivation of multiplicative update rules for nonnegative matrix factorization with regularization terms," Proceedings of the 2017 International Symposium on Nonlinear Theory and Its Applications, pp.322-325, 2017.

Reply:
cf. https://gist.github.com/tsano430/dc005fd59f89662d777774de84cd5d0d